src/extension/sdk/utils.ts

```diff
@@ -237,22 +239,38 @@ export class SdkUtils {
 				this.logger.info(`Found Fuchsia project that is not vanilla Flutter`);
 		}
 
-		const flutterSdkSearchPaths = [
-			workspaceConfig?.flutterSdkHome,
-			config.flutterSdkPath,
-			// TODO: These could move into processFuchsiaWorkspace and be set on the config?
-			fuchsiaRoot && path.join(fuchsiaRoot, "lib/flutter"),
-			fuchsiaRoot && path.join(fuchsiaRoot, "third_party/dart-pkg/git/flutter"),
-			firstFlutterMobileProject,
-			firstFlutterMobileProject && extractFlutterSdkPathFromPackagesFile(path.join(firstFlutterMobileProject, ".packages")),
-			firstFlutterMobileProject && path.join(firstFlutterMobileProject, ".flutter"),
-			firstFlutterMobileProject && path.join(firstFlutterMobileProject, "vendor/flutter"),
-			process.env.FLUTTER_ROOT,
-		].concat(paths).filter(notUndefined);
-
-		const flutterSdkPath = this.findFlutterSdk(flutterSdkSearchPaths);
-		// Since we just blocked on a lot of sync FS, yield.
-		await resolvedPromise;
+		let flutterSdkPath;
+		if (fs.existsSync(flutterSnapBinPath)) {
+			flutterSdkPath = flutterSnapSdkPath;
+			// If the Flutter snap is installed but not initialised, initialise it now.
+			if (!fs.existsSync(flutterSnapSdkPath + "/.git")) {
+				await window.withProgress(
```

Comment:
I'm thinking we might want to ask the user first before starting this. Although they're unlikely to say no, it really bugs me when a VS Code extension just starts downloading (potentially large) files without any prompt - especially when VS Code can eagerly activate extensions for many unexpected reasons (for example if you open a large non-Dart project and VS Code fails to walk the tree within 7 seconds, it will [automatically activate the Dart extension](https://github.com/microsoft/vscode/issues/73656)!)